--- gda-export.c	2004-11-13 00:25:44.396760885 -0600
+++ libgda-1.0.0/libgda/gda-export.c	2004-11-13 00:23:58.405595352 -0600
@@ -213,10 +213,10 @@
 			      G_TYPE_NONE, 2,
 			      G_TYPE_INT, G_TYPE_STRING);
 	gda_export_signals[CANCELLED] =
-		g_signal_new ("finished",
+		g_signal_new ("cancelled",
 				G_TYPE_FROM_CLASS (object_class),
 			      G_SIGNAL_RUN_LAST,
-			      G_STRUCT_OFFSET (GdaExportClass, finished),
+			      G_STRUCT_OFFSET (GdaExportClass, cancelled),
 			      NULL, NULL,
 			      g_cclosure_marshal_VOID__VOID,
 			      G_TYPE_NONE, 0);
