--- gnome-db-list.c	2004-11-13 00:01:42.391981026 -0600
+++ libgnomedb-1.0.0/libgnomedb/gnome-db-list.c	2004-11-13 00:02:10.172354404 -0600
@@ -111,7 +111,7 @@
 		g_signal_new ("selection_cleared",
 			      G_TYPE_FROM_CLASS (object_class),
 			      G_SIGNAL_RUN_LAST,
-			      G_STRUCT_OFFSET (GnomeDbGridClass, row_selected),
+			      G_STRUCT_OFFSET (GnomeDbGridClass, selection_cleared),
 			      NULL, NULL,
 			      g_cclosure_marshal_VOID__VOID,
 			      G_TYPE_NONE, 0);
