--- gnome-db-list.c	2004-11-13 00:01:42.391981026 -0600
+++ libgnomedb-1.0.0/libgnomedb/gnome-db-list.c	2004-11-13 00:02:10.172354404 -0600
@@ -111,7 +111,7 @@
 		g_signal_new ("selection_cleared",
 			      G_TYPE_FROM_CLASS (object_class),
 			      G_SIGNAL_RUN_LAST,
-			      G_STRUCT_OFFSET (GnomeDbGridClass, row_selected),
+			      G_STRUCT_OFFSET (GnomeDbGridClass, selection_cleared),
 			      NULL, NULL,
 			      g_cclosure_marshal_VOID__VOID,
 			      G_TYPE_NONE, 0);
--- libgnomedb-1.0.0/libgnomedb/gnome-db-editor.h.orig	2005-02-23 13:50:59.509288486 -0500
+++ libgnomedb-1.0.0/libgnomedb/gnome-db-editor.h	2005-02-23 13:50:51.247345662 -0500
@@ -29,7 +29,7 @@
 
 #define GNOME_DB_TYPE_EDITOR            (gnome_db_editor_get_type())
 #define GNOME_DB_EDITOR(obj)            (G_TYPE_CHECK_INSTANCE_CAST (obj, GNOME_DB_TYPE_EDITOR, GnomeDbEditor))
-#define GNOME_DB_EDITOR_CLASS(klass)    (G_TYPE_CHECK_CLASS_CAST (klass, GNOME_DB_TYPE_EDITOR, GnomeDbEditor))
+#define GNOME_DB_EDITOR_CLASS(klass)    (G_TYPE_CHECK_CLASS_CAST (klass, GNOME_DB_TYPE_EDITOR, GnomeDbEditorClass))
 #define GNOME_DB_IS_EDITOR(obj)         (G_TYPE_CHECK_INSTANCE_TYPE (obj, GNOME_DB_TYPE_EDITOR))
 #define GNOME_DB_IS_EDITOR_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), GNOME_DB_TYPE_EDITOR))
 
