--- gtk+-3.16.6.orig/gtk/gtktextview.h	2015-09-04 11:37:38.035020919 +0200
+++ gtk+-3.16.6/gtk/gtktextview.h	2015-09-04 11:38:59.958096121 +0200
@@ -155,8 +155,6 @@
 {
   GtkContainerClass parent_class;
 
-  /*< public */
-
   void (* populate_popup)        (GtkTextView      *text_view,
                                   GtkWidget        *popup);
   void (* move_cursor)           (GtkTextView      *text_view,
