--- gtkclipboard.h.orig	2005-03-18 15:41:55.748377369 -0500
+++ gtkclipboard.h	2005-03-18 15:45:36.752348627 -0500
@@ -32,6 +32,9 @@
 #define GTK_CLIPBOARD(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_TYPE_CLIPBOARD, GtkClipboard))
 #define GTK_IS_CLIPBOARD(obj)         (G_TYPE_CHECK_INSTANCE_TYPE ((obj), GTK_TYPE_CLIPBOARD))
 
+typedef struct _GtkClipboard GtkClipboard;
+typedef struct _GtkClipboardClass GtkClipboardClass;
+
 typedef void (* GtkClipboardReceivedFunc)        (GtkClipboard     *clipboard,
 					          GtkSelectionData *selection_data,
 					          gpointer          data);
