diff -Naur gtk+-3.22.1.orig/gtk/gtktreemodel.c gtk+-3.22.1/gtk/gtktreemodel.c
--- gtk+-3.22.1.orig/gtk/gtktreemodel.c	2016-08-29 18:20:43.000000000 +0100
+++ gtk+-3.22.1/gtk/gtktreemodel.c	2016-10-17 13:57:41.204889300 +0100
@@ -395,14 +395,15 @@
       closure = g_closure_new_simple (sizeof (GClosure), NULL);
       g_closure_set_marshal (closure, row_inserted_marshal);
       tree_model_signals[ROW_INSERTED] =
-        g_signal_newv (I_("row-inserted"),
+        g_signal_new (I_("row-inserted"),
                        GTK_TYPE_TREE_MODEL,
                        G_SIGNAL_RUN_FIRST,
-                       closure,
+                       G_STRUCT_OFFSET (GtkTreeModelIface, row_inserted),
                        NULL, NULL,
                        _gtk_marshal_VOID__BOXED_BOXED,
                        G_TYPE_NONE, 2,
-                       row_inserted_params);
+                       GTK_TYPE_TREE_PATH | G_SIGNAL_TYPE_STATIC_SCOPE,
+                       GTK_TYPE_TREE_ITER);
 
       /**
        * GtkTreeModel::row-has-child-toggled:
@@ -441,14 +442,14 @@
       closure = g_closure_new_simple (sizeof (GClosure), NULL);
       g_closure_set_marshal (closure, row_deleted_marshal);
       tree_model_signals[ROW_DELETED] =
-        g_signal_newv (I_("row-deleted"),
+        g_signal_new (I_("row-deleted"),
                        GTK_TYPE_TREE_MODEL,
                        G_SIGNAL_RUN_FIRST,
-                       closure,
+                       G_STRUCT_OFFSET (GtkTreeModelIface, row_deleted),
                        NULL, NULL,
                        NULL,
                        G_TYPE_NONE, 1,
-                       row_deleted_params);
+                       GTK_TYPE_TREE_PATH);
 
       /**
        * GtkTreeModel::rows-reordered: (skip)
@@ -471,14 +472,15 @@
       closure = g_closure_new_simple (sizeof (GClosure), NULL);
       g_closure_set_marshal (closure, rows_reordered_marshal);
       tree_model_signals[ROWS_REORDERED] =
-        g_signal_newv (I_("rows-reordered"),
+        g_signal_new (I_("rows-reordered"),
                        GTK_TYPE_TREE_MODEL,
                        G_SIGNAL_RUN_FIRST,
-                       closure,
+                       G_STRUCT_OFFSET (GtkTreeModelIface, rows_reordered),
                        NULL, NULL,
                        _gtk_marshal_VOID__BOXED_BOXED_POINTER,
                        G_TYPE_NONE, 3,
-                       rows_reordered_params);
+                       GTK_TYPE_TREE_PATH | G_SIGNAL_TYPE_STATIC_SCOPE,
+                       GTK_TYPE_TREE_ITER, G_TYPE_POINTER);
       initialized = TRUE;
     }
 }
